R/transformCounts.R

```diff
@@ -779,6 +793,52 @@ setMethod("transformAssay", signature = c(x = "SingleCellExperiment"),
     return(mat)
 }
 
+########################### .apply_transformation_difference ##################
+# Computes all pairwise differences (x - y) between features across samples.
+# Returns a matrix with one row per feature pair.
+.apply_transformation_difference <- function(mat, ...) {
+  # Get the feature (taxa) names
+  taxa <- rownames(mat)
+  
+  # Generate all unique pairwise combinations of taxa
+  combs <- utils::combn(taxa, 2, simplify = FALSE)
+  
+  # Compute pairwise differences: x - y for each taxa pair across samples
+  diffs <- lapply(combs, function(pair) {
+    t1 <- pair[1]; t2 <- pair[2]
+    name <- paste0("diff_", t1, "_", t2)
+    values <- mat[t1, ] - mat[t2, ]
+    matrix(values, nrow = 1, dimnames = list(name, colnames(mat)))
+  })
+  
+  # Combine all difference rows into a single matrix
+  result <- do.call(rbind, diffs)
+  
+  # Add metadata
+  attr(result, "mia") <- "diff"
+  
+  return(result)
+}
+
+############################ .apply_transformation_division ############################
+# Computes all pairwise ratios (x / y) between features across samples.
+# Returns a matrix with one row per feature pair.
+.apply_transformation_division <- function(mat, ...) {
+  taxa <- rownames(mat)
+  combs <- utils::combn(taxa, 2, simplify = FALSE)
+  
+  ratios <- lapply(combs, function(pair) {
+    t1 <- pair[1]; t2 <- pair[2]
+    name <- paste0("div_", t1, "_over_", t2)
+    values <- mat[t1, ] / mat[t2, ]
+    matrix(values, nrow = 1, dimnames = list(name, colnames(mat)))
+  })
+  
+  result <- do.call(rbind, ratios)
+  attr(result, "mia") <- "division"
+  return(result)
+}
+
 # This function is used to add transformed table back to TreeSE. With most of
 # the methods it is simple: it is added to assay. However, with philr, the
```

Comment:
not only with philr but also with difference and division - you could change the text as "However, with transformations that change the dimensionality (e.g. philr, difference, division), the transformed table is added to altExp instead." (or if it is just returned as a matrix rather than added as altExp then say that, I didnt check yet which one this does)